### PR TITLE
Add TopicBlogLauncher edit and add

### PR DESCRIPTION
**Pull request by mcmikashi**
I implement the form I reoder the fields but don't know if this good or maybe split like topicblog item ?

And for need some detail for the template config for this object.

### CODE REVIEW

**Comment by JeffAbrahamson** [comment]
Can template_name be a dropdown of available templates like for TBItem?

**Comment by JeffAbrahamson** [comment]
Looks very good overall!

**Comment by mcmikashi** [comment]
- I reorder the fields
- I add template config on TB launcher model
- The template name  field is now a select field

**Review by Shriukan33 — CHANGES_REQUESTED** [review]
Attempting to edit and save an existing item will raise an Error : 
image

The ImageField being mandatory, it's not transferred between items. We already had this problem and solved it here : 
https://github.com/transport-nantes/tn_web/blob/bf005527ab8ad95ec6524cf02fc189e4b5286a61/transport_nantes/topicblog/views.py#L289

**Comment by JeffAbrahamson** [comment]
In bf005527, please be careful of non-empty blank lines.  Your editor should have support to flag them to you.  Certainly `git diff` does.

image

**Comment by JeffAbrahamson** [comment]
When you copied and pasted the comments about content type, did you think about what they mean?  I'm pretty sure this comment in TBItem is obsolete and has no meaning, although the code that follows does have meaning.

So what concerns me is not that you copied and pasted: these classes are pretty parallel, and so that makes sense to get yourself started.  What concerns me is that you may have copied and pasted without reading and understanding.

The code that follows the comment looks correct.

File: transport_nantes/topicblog/models.py [hunk 1]
```diff
@@ -840,6 +840,45 @@ class Meta:
...
+    # "content type": [list of templates available for this type]
+    # The templates in the list must :
+    # 1) Exist in the templates/ directory
+    # 2) be configurated in self.template_config
+    # Default values for template_config ###########################
+    template_config_default = {
+        "optional_fields_for_publication": (),
+        # Fields that, if required for publication, the requirement is
+        # satisfied by providing any one of them.
+        "one_of_fields_for_publication": [['launcher_image',
```
[the remark this answers is not in the document]
Comment (JeffAbrahamson) [reply transport_nantes/topicblog/models.py:855]: Yes, I agree, either you provide an image and alt, or you provide neither, but not just one.

**Comment by mcmikashi** [comment]
Before reviewing this PR, I need this PR to be reviewed  [PR 538](https://github.com/transport-nantes/tn_web/pull/538)

- I move the function get_image_fields from TopicBlogItem to TopicBlogOjectBase
- I add template_config in TopicBlogLauncher
- I add form_post_process in TopicBlogLauncherEdit
- I add edit test for topicblog launcher

The test fails because when I test topicblog:edit_launcher_by_slug
with a bad slug, I got a 200 code status.
What should happen when we get a bad slug on this page?

- Redirect to add ?
- Let the user edit ? (have to rewrite the slug)
- Let the user edit but the slug field is prefilled?

and what should happen if the user gives a good slug?

- let him edit the last published item?
- or let him edit the last added?

**Comment by JeffAbrahamson** [comment]
Please mark PRs as drafts if tests are failing or they aren't otherwise ready for review.  The reviewers job is not to re-test your code but rather to make sure you're more or less going about your job correctly.  That is, they're offering a second set of eyes, but they aren't checking that you know how to do your job at all!

**Comment by mcmikashi** [comment]
I have a little problem with edit with topicblog:edit_launcher_by_slug when we got the edit permission it’s always return a 200 status code even if we give a bad slug and with a good slug 
the fields stay empty.
Maybe we can relate this to the issue #552 or create a new one ?

**Comment by JeffAbrahamson** [comment]
@mcmikashi You've unmarked this from being a draft, but you are indicating in your last comment that it's not ready?

**Comment by JeffAbrahamson** [comment]
LGTM overall.
Tests pass, so I'm merging this.

**Pull request closed without merging**